packages/logger/tests/unit/injectLambdaContext.test.ts

```diff
@@ -239,4 +240,333 @@ describe('Inject Lambda Context', () => {
       );
     }
   );
+
+  describe('Correlation ID', () => {
+    const testEvent = {
+      headers: {
+        'x-correlation-id': '12345-test-id',
+      },
+      requestContext: {
+        requestId: 'api-gateway-request-id',
+      },
+      id: 'eventbridge-id',
+    };
+
+    beforeEach(() => {
+      process.env = {
+        ...ENVIRONMENT_VARIABLES,
+        POWERTOOLS_DEV: 'true',
+      };
+      vi.clearAllMocks();
+    });
+
+    it.each([
+      {
+        case: 'middleware',
+        getHandler: (logger: Logger) =>
+          middy(async () => {
+            logger.info('Hello, world!');
+          }).use(
+            injectLambdaContext(logger, {
+              correlationIdPath: 'headers."x-correlation-id"',
+            })
+          ),
+      },
+      {
+        case: 'decorator',
+        getHandler: (logger: Logger) => {
+          class Lambda {
+            @logger.injectLambdaContext({
+              correlationIdPath: 'headers."x-correlation-id"',
+            })
+            public async handler(
+              _event: unknown,
+              _context: Context
+            ): Promise<void> {
+              logger.info('Hello, world!');
+            }
+          }
+          const lambda = new Lambda();
+          return lambda.handler.bind(lambda);
+        },
+      },
+    ])('should set correlation ID through $case', async ({ getHandler }) => {
+      // Prepare
+      const logger = new Logger({ correlationIdSearchFn: search });
+      const handler = getHandler(logger);
+
+      // Act
+      await handler(testEvent, context);
+
+      // Assess
+      expect(console.info).toHaveBeenCalledTimes(1);
+      expect(console.info).toHaveLoggedNth(
+        1,
+        expect.objectContaining({
+          message: 'Hello, world!',
+          correlation_id: '12345-test-id',
+          ...getContextLogEntries(),
+        })
+      );
+      expect(logger.getCorrelationId()).toBe('12345-test-id');
+    });
+
+    it('should set correlation ID manually', () => {
+      // Prepare
+      const logger = new Logger();
+
+      // Act
+      logger.addContext(context);
+      logger.setCorrelationId('12345-test-id');
+      logger.info('Hello, world!');
+
+      // Assess
+      expect(console.info).toHaveBeenCalledTimes(1);
+      expect(console.info).toHaveLoggedNth(
+        1,
```

Comment:
With the current implementation, this is the same test as a couple down below - they're just aliases to `setAttribute()`